### ISSUE TITLE
linter: jsx-a11y/click-events-have-key-events ignores settings.jsx-a11y.attributes href alias for component-mapped elements

### ISSUE THREAD

**Issue by JCast867** [oxc-project/oxc#25942]
### What version of Oxlint are you using?

1.79.0

### What command did you run?

oxlint -c .oxlintrc.json src

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["jsx-a11y"],
  "settings": {
    "jsx-a11y": {
      "components": { "Link": "a" },
      "attributes": { "href": ["to"] }
    }
  },
  "rules": {
    "jsx-a11y/click-events-have-key-events": "error"
  }
}

```

### What happened?

`jsx-a11y/click-events-have-key-events` flags a component mapped to `<a>` via `settings.jsx-a11y.components`, even when the component's href-equivalent prop is declared via `settings.jsx-a11y.attributes`.

`get_element_type` does appear to consult `settings.jsx-a11y.components` — I confirmed this from the existing `<Footer>` → `"footer"` test case in `click_events_have_key_events.rs`, so a component like React Router's `<Link>` correctly resolves to element type `"a"`. But the subsequent interactivity check (which is supposed to exempt `<a href="...">`) doesn't seem to also consult `settings.jsx-a11y.attributes` to recognize an aliased prop (e.g. `to`) as equivalent to a literal `href` — so `<Link to="...">` still gets flagged as if it were an anchor with no `href` at all.

Repro:

```tsx
import { Link } from "react-router";

function Example() {
  return (
    <Link to="/somewhere" onClick={() => {}}>
      Go
    </Link>
  );
}
```

Expected: no error, since `Link` resolves to `"a"` via `components`, and `to` should resolve to the `href` equivalent via attributes. The same as `<a href="/somewhere" onClick={...}>`, which is exempt.

Actual: `click-events-have-key-events` still fires, requiring an `onKeyUp`/`onKeyDown`/`onKeyPress` handler.

This behavior matches the original `eslint-plugin-jsx-a11y`, where both `components` and `attributes` settings are consulted together by shared utilities across rules including this one — so this looks like a gap specific to Oxlint's Rust port of `is_interactive_elemen`t (or wherever the href check lives) not reading the `attributes` alias for anchor interactivity, even though `components` mapping is honored.

**Comment by AmalJossy** [comment]
`eslint-plugin-jsx-a11y` also returns an error for the repro. Not sure if there is a gap. 

> This behavior matches the original eslint-plugin-jsx-a11y, where both components and attributes settings are consulted together by shared utilities across rules including this one

This [code search](https://github.com/search?q=repo%3Ajsx-eslint%2Feslint-plugin-jsx-a11y+%22settings%5B%27jsx-a11y%27%5D%22&type=code) says the `attributes `settings are consulted only for 2 rules and not in any shared utils
